client/src/components/Ticketing/ticketingmanager/TicketTypes/SubscriptionTypes.tsx

```diff
@@ -0,0 +1,374 @@
+      setSubscriptionTypes((prev) => prev.filter((type) => type.id !== -1));
+      return prevRow;
+    }
+  };
+
+  const addRow = () => {
+    setSubscriptionTypes((types) => [...types, getDefaultSubscriptionType()]);
+    setRowModesModel({...rowModesModel, [-1]: {mode: GridRowModes.Edit}});
+    setEditing(true);
+  };
+
+  const columns: GridColumns = [
+    {
+      field: 'name',
+      headerName: 'Subscription Type',
+      width: 130,
+      editable: true,
+      preProcessEditCellProps: (params: GridPreProcessEditCellProps) => ({
+        ...params.props,
+        error:
+          !params.props.value || params.props.value.length === 0
+            ? 'Invalid Input'
+            : undefined,
+      }),
+      renderEditCell: EditCellWithTooltip,
+    },
+    {
+      field: 'description',
+      headerName: 'Description',
+      width: 400,
+      editable: true,
+      preProcessEditCellProps: (params: GridPreProcessEditCellProps) => ({
+        ...params.props,
+        error:
+          !params.props.value || params.props.value.length === 0
+            ? 'Invalid Input'
+            : undefined,
+      }),
+      renderEditCell: EditCellWithTooltip,
+    },
+    {
+      field: 'price',
+      headerName: 'Default Price',
+      width: 110,
+      editable: true,
+      preProcessEditCellProps: (params: GridPreProcessEditCellProps) => ({
+        ...params.props,
+        error:
+          isNaN(+params.props.value) || +params.props.value <= 0
+            ? 'Invalid Input'
+            : undefined,
+      }),
+      valueFormatter: (params: GridValueFormatterParams) =>
+        formatUSD(+params.value),
+      renderEditCell: EditCellWithTooltip,
+    },
+    {
+      field: 'previewonly',
+      headerName: 'Preview Only',
+      width: 110,
+      editable: true,
+      renderCell: (cell) => {
+        return (
+          <div className='w-full grid justify-center'>
+            <Switch
+              checked={cell.value}
+              color='primary'
+              size='small'
+              disabled={true}
+            />
+          </div>
+        );
+      },
+      renderEditCell: (cell) => {
+        const apiRef = useGridApiContext();
+        return (
+          <div className='w-full grid justify-center'>
+            <Switch
+              checked={cell.value}
+              onChange={() =>
+                apiRef.current.setEditCellValue({
+                  id: cell.id,
+                  field: cell.field,
+                  value: !cell.value,
+                })
+              }
+              color='primary'
+              size='small'
+            />
+          </div>
+        );
+      },
+    },
+    {
+      field: 'actions',
+      headerName: '',
+      sortable: false,
+      width: 100,
+      renderCell: (cell) => {
+        return rowModesModel[cell.id]?.mode === GridRowModes.Edit ? (
+          <div className='w-[100%] flex flex-row justify-around'>
```

Comment:
For consistency, do you think you could mirror these buttons after the same ones from the Ticket Types screen (styling, placement, and icons)

Reply:
Updated